proxy/coco/src/peer/run_state.rs

```diff
@@ -299,7 +301,7 @@ pub struct RunState {
     /// Confiugration to change how input [`Input`]s are interpreted.
     config: Config,
     /// Tracking remote peers that have an active connection.
-    connected_peers: HashSet<PeerId>,
+    connected_peers: HashMap<PeerId, usize>,
```

Comment:
I'm not sure I understood the motivation for the `HashMap`. Could you educate me, please? :)

Reply:
The peer can receive multiple connections from the same `peer_id`,  a vivid example is the extra connection that is established during syncs (git only exchange). If we just keep track of simply a binary connected status the following order of events can leave the peer in an offline state: `Connect(Peer1) -> Connect(Peer1) -> Disconnect(Peer1)`.

Reply:
Ahhhh very good :clap: Can you expand on the docs for `connected_peers` to outline this case? Some prose like that would have illuminated my mind immediately :bow:

Reply:
1c73907